File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(graph , visited, v):
    visited[v] = True
    for i in graph[v]:
        if not visited[i]:
            dfs(graph , visited, i)
    return 1

def solution(n, computers):
    answer = 0
    graph = [[] for _ in range(n+1)]
    for i in range(1,n+1):
        for j in range(1, n+1):
            if computers[i-1][j-1] == 1 and i!= j:
                if j not in graph[i]:
                    graph[i].append(j)
                    graph[j].append(i)
    # dfs > stack
    visited = [False] * (n+1)
    
    for start in range(1, n + 1):
        if not visited[start]:
            answer += dfs(graph , visited, start)

    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    boolean visited[];
    Queue<Integer> queue = new LinkedList<>();
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[computers.length];
        
        for(int i = 0; i < computers.length; i++){
            if(visited[i] == false){
                bfs(i, computers, computers.length);
                answer++;
            }
        }
        return answer;
    }
    
    void bfs(int i, int computers[][], int n){
        queue.offer(i); // 이게 여기선 append인가봐 신기하다..
        visited[i] = true;
        
        while(!queue.isEmpty()){
            int v = queue.poll();
            
            for (int j = 0; j < n; j++){
                if (visited[j] == false && computers[v][j] == 1){
                    visited[j] = true;
                    queue.offer(j);
                }
            }
        }
    }
}